e2e/tests/functional/plugins/plot/barGraph.e2e.spec.js

```diff
@@ -0,0 +1,86 @@
+ * Administration. All rights reserved.
+ *
+ * Open MCT is licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+/*
+ * This test suite is dedicated to testing the Bar Graph component.
+ */
+
+import { v4 as uuid } from 'uuid';
+
+import { createDomainObjectWithDefaults } from '../../../../appActions.js';
+import { expect, test } from '../../../../pluginFixtures.js';
+
+test.describe('Bar Graph', () => {
+  let barGraph;
+
+  test.beforeEach(async ({ page }) => {
+    // Open a browser, navigate to the main page, and wait until all networkevents to resolve
+    await page.goto('./', { waitUntil: 'domcontentloaded' });
+
+    // Create the Bar Graph
+    barGraph = await createDomainObjectWithDefaults(page, { type: 'Graph' });
+  });
+
+  test('Requests a single historical datum', async ({ page }) => {
+    // Create a sine wave generator within the bar graph
+    await createDomainObjectWithDefaults(page, {
+      type: 'Sine Wave Generator',
+      name: `swg-${uuid()}`,
+      parent: barGraph.uuid
+    });
+    const getRequestOptionPromise = addTelemetryInterceptor(page);
+
+    // Navigate to the bar graph and verify that
+    await page.goto(barGraph.url, { waitUntil: 'domcontentloaded' });
+    await page.getByRole('tab', { name: 'Config' }).click();
+
+    const requestOption = await getRequestOptionPromise;
+    await expect(requestOption).toBe('latest');
+  });
+});
+
+/**
+ * Util for returning the size option for telemetry requests
+ * @param {import('@playwright/test').Page} page
+ */
+async function addTelemetryInterceptor(page) {
+  const getRequestOptionPromise = new Promise((resolve) =>
+    page.exposeFunction('getRequestOption', resolve)
+  );
+
+  await page.evaluate(() => {
+    function requestOptionsChecker() {
+      return {
+        appliesTo: () => {
+          return true;
+        },
+        invoke: (request) => {
+          window.getRequestOption(request.strategy);
+          return request;
+        }
+      };
+    }
+    window.openmct.telemetry.addRequestInterceptor(requestOptionsChecker());
+  });
```

Comment:
i worry about this introducing flake-- will test this locally.

Is this to avoid the need for a couchdb test?

Reply:
It's because we didn't have any way to make network requests